sdks/python/apache_beam/utils/subprocess_server.py

```diff
@@ -161,8 +161,9 @@ class JavaJarServer(SubprocessServer):
   BEAM_GROUP_ID = 'org.apache.beam'
   JAR_CACHE = os.path.expanduser("~/.apache_beam/cache/jars")
 
-  _BEAM_SERVICES = threading.local()
-  _BEAM_SERVICES.replacements = {}
+  _BEAM_SERVICES = _BEAM_SERVICES = type(
```

Comment:
Additional `_BEAM_SERVICES`?

It seems like creating a subclass of `threading.local()` would work in this case. Why `type` also works?

Reply:
Hmm... not sure how the duplicate got in there. 

`type(...)` is actually creating a subclass here.

Reply:
Thanks! I'll merge it after all tests pass.